libs/execution_context/ebpf_maps.c

```diff
@@ -866,6 +866,84 @@ _update_entry_per_cpu(
     return EBPF_SUCCESS;
 }
 
+// Given two keys, compute the number of bits that match.
+static size_t
+_lpm_trie_matching_key_prefix_length(
+    _In_reads_((key_length_a + 7) / 8) const uint8_t* key_a,
+    size_t key_length_a,
+    _In_reads_((key_length_b + 7) / 8) const uint8_t* key_b,
+    size_t key_length_b)
+{
+    size_t maximum_matching_length = key_length_a > key_length_b ? key_length_b : key_length_a;
+    size_t remaining_key_length_in_bits = maximum_matching_length;
+    uint8_t key_a_last_byte;
+    uint8_t key_b_last_byte;
+    size_t index;
+
+    while (remaining_key_length_in_bits >= 8) {
+        index = (maximum_matching_length - remaining_key_length_in_bits) / 8;
+        if (key_a[index] != key_b[index]) {
+            break;
+        }
+        remaining_key_length_in_bits -= 8;
+    }
+    if (!remaining_key_length_in_bits) {
+        return maximum_matching_length;
+    }
+
+    index = (maximum_matching_length - remaining_key_length_in_bits) / 8;
+    key_a_last_byte = key_a[index];
+    key_b_last_byte = key_b[index];
+
+    while (remaining_key_length_in_bits) {
+        if ((key_a_last_byte & 0x80) != (key_b_last_byte & 0x80)) {
```

Comment:
Do a XOR and find the first set bit by calling https://docs.microsoft.com/en-us/cpp/intrinsics/bitscanforward-bitscanforward64?view=msvc-160? Probably same complexity. NM.